### PR TITLE
[Android Authorization QS] Use IdToken instead of AppMetadata to set Role

### PR DESCRIPTION
**Pull request by lbalmaceda**
@jeffreylees Please review the URLs changed in this article as I'm not familiar with doc's syntax.

### CODE REVIEW

File: articles/quickstart/native/android/07-authorization.md [hunk 1]
```diff
@@ -20,56 +20,58 @@ This step demonstrates how to use Auth0 to create access roles for your users. W
...
+}
 ```
-to match the condition that fits your needs.
 
-By default, it says that if the user email contains `@example.com` then they will be given an `admin` role, otherwise a regular `user` role.
+The rule will run every time a user attempts to authenticate. If the user has a valid `email` and the email's domain is `admin.com`, the user will be granted the `user` and `admin` roles. On any other case, the user will be granted a simple `user` role. This new claim is saved in the *id_token* under the `https://access.control/roles` name, which you will have to check on your app.
 
 ::: note
-You can define more roles other than `admin` and `user`, depending on your product requirements.
+Feel free to customize the required conditions and the different roles to grant on each of them. You can define more roles other than `admin` and `user`, depending on your product requirements. There is a restriction on the naming of the claims, please read [this article](https://auth0.com/docs/rules/current#hello-world) for more context about Rules.
```
Comment (jeffreylees) [inline articles/quickstart/native/android/07-authorization.md]: The link can be shortened here. I might reword this note to be like:

`Feel free to customize the conditions and the roles to grant on each of them. You can define as many roles as your project requires, beyond `admin` and `user`. There are some restrictions on the naming of the claims detailed in the [rules documentation](https://auth0.com/docs/rules/current#hello-world).`

File: articles/quickstart/native/android/07-authorization.md [hunk 2]
```diff
@@ -20,56 +20,58 @@ This step demonstrates how to use Auth0 to create access roles for your users. W
...
-  };
-```
 
 ## Test the Rule in Your Project
 
-Once you have the user profile (as explained in the [user profile](04-user-profile) tutorial), you can save it and access it at any point.
+Once you have the user credentials (as explained in the [Login](/quickstart/native/android/00-login) tutorial), you can save and access them at any point.
 
-Inside it, you will have the role, and you will be ready to perform the access control.
+The *id_token* is a [JWT](https://auth0.com/docs/jwt) that holds several claims, like the one with the roles you set. By using a *JWT decoding library* like [this one](https://github.com/auth0/JWTDecode.Android) you will be able to obtain the roles and perform the access control.
```
Comment (jeffreylees) [inline articles/quickstart/native/android/07-authorization.md]: Shortened URL, and removing the italics - apparently we're doing away with using italics for emphasis most places in /docs.

`The id_token is a [JWT](/jwt) that holds several claims, like the one with the roles you set previously. By using a JWT decoding library (like [this one](https://github.com/auth0/JWTDecode.Android)) you will be able to obtain the roles and perform access control.`

**Review by jeffreylees — CHANGES_REQUESTED** [review]
All good other than a couple of URL changes

File: articles/quickstart/native/android/07-authorization.md (hunk 1 above)
Comment (mpaktiti) [reply articles/quickstart/native/android/07-authorization.md]: And this `https://auth0.com/docs/rules/current#hello-world` should be `/rules#hello-world`

**Comment by lbalmaceda** [comment]
Done 🎉

File: articles/quickstart/native/android/07-authorization.md [hunk 3]
```diff
@@ -50,15 +50,15 @@ function (user, context, callback) {
 The rule will run every time a user attempts to authenticate. If the user has a valid `email` and the email's domain is `admin.com`, the user will be granted the `user` and `admin` roles. On any other case, the user will be granted a simple `user` role. This new claim is saved in the *id_token* under the `https://access.control/roles` name, which you will have to check on your app.
 
 ::: note
-Feel free to customize the required conditions and the different roles to grant on each of them. You can define more roles other than `admin` and `user`, depending on your product requirements. There is a restriction on the naming of the claims, please read [this article](https://auth0.com/docs/rules/current#hello-world) for more context about Rules.
+Feel free to customize the conditions and the roles to grant on each of them. You can define as many roles as your project requires, beyond `admin` and `user`. There are some restrictions on the naming of the claims detailed in the [rules documentation](/rules/#hello-world).
```
Comment (mpaktiti) [inline articles/quickstart/native/android/07-authorization.md]: @lbalmaceda The URL should be `[rules documentation](/rules#hello-world)`

**Comment by lbalmaceda** [comment]
Rebased

File: articles/quickstart/native/android/07-authorization.md [hunk 4]
```diff
@@ -21,61 +21,57 @@ This tutorial shows you how to use Auth0 to create access roles for your users.
...
+}
 ```
 
-The default rules for assigning access roles are:
-* If the user's email contains `@example.com`, the user gets the admin role. If the email contains anything else, the user gets the regular user role.
+Rules run every time a user authenticates. If the user has a valid `email` and the email's domain is `admin.com`, the user will be granted the `user` and `admin` roles. On any other case, the user will be granted a simple `user` role. The user's role value is saved in the id token under the `https://access.control/roles` claim, which has to be checked on the app.
 
 ::: note
-Depending on your needs, you can define roles other than admin and user.
+The rule can be customized to grant the user different roles other than the ones explained here, depending on the conditions required. The name of the claims added to the id token must be name-spaced (look like a URL). Read [this article](https://auth0.com/docs/rules/current#hello-world) for more context about Rules.
```
Comment (jeffreylees) [inline articles/quickstart/native/android/07-authorization.md]: use non absolute link + remove the subsection: `[this article](/rules/current)`

File: articles/quickstart/native/android/07-authorization.md [hunk 5]
```diff
@@ -21,61 +21,57 @@ This tutorial shows you how to use Auth0 to create access roles for your users.
...
-  };
-```
 
 ## Test the Rule in Your Project
 
-Once you have the user profile, you can save the profile and access it at any point. To learn more about user profiles, see the [User Profile](04-user-profile) tutorial. 
+Once the user credentials had been obtained (as explained in the [Login](/quickstart/native/android/00-login) tutorial), save them to access them at any time.
 
-The user profile contains the access role assigned to the user. 
+The id token is a [JSON Web Token](https://auth0.com/docs/jwt) that holds several claims, like the one with the roles set on this tutorial. Use a *JWT decoding library* like [this one](https://github.com/auth0/JWTDecode.Android) to obtain the roles and perform the access control.
```
Comment (jeffreylees) [inline articles/quickstart/native/android/07-authorization.md]: Short link: `[JSON Web Token](/jwt)`

**Review by jeffreylees — CHANGES_REQUESTED** [review]
A couple of other link adjustments and I'm fine with this

**Comment by lbalmaceda** [comment]
@jeffreylees done 👍

**Review by jeffreylees — APPROVED** [review]

**Comment by lbalmaceda** [comment]
Someone please merge this PR

**Review by mpaktiti — APPROVED** [review]

**Pull request merged**